src/controllers/HistoryController.ts

```diff
@@ -17,22 +17,31 @@ import {Utils} from '../utils/Utils';
 export class HistoryController extends RootComponent {
   static ID = 'HistoryController';
 
+  static HashUtilsCopy: typeof hashUtils.HashUtils = _.clone(hashUtils.HashUtils);
   static attributesThatDoNotTriggerQuery = ['quickview'];
 
   private ignoreNextHashChange = false;
   private initialHashChange = false;
   private willUpdateHash: boolean = false;
   private hashchange: (...args: any[]) => void;
+  private hashUtils: typeof hashUtils.HashUtils;
 
   /**
    * Create a new history controller
    * @param element
-   * @param windoh For mock / test purpose.
+   * @param windoh For mock / test purposes.
    * @param model
    * @param queryController
+   * @param hashUtilsModule For mock / test purposes.
    */
-  constructor(element: HTMLElement, public windoh: Window, public model: Model, public queryController: QueryController) {
+  constructor(element: HTMLElement, public windoh: Window, public model: Model, public queryController: QueryController, hashUtilsModule?: typeof hashUtils.HashUtils) {
```

Comment:
As discussed live, you can add a constructor param for the module itself. You can check the Debug class